### PR TITLE
Allow passing in stdin and custom compiler arguments

### PR DESCRIPTION
**Pull request by wilzbach**
Ported https://github.com/dlang-tour/core-rdmd/pull/9

Amazingly the testsuite worked - let's see whether it works for all images as well (except gdc).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,10 @@ sudo: required
 services:
```
Comment (PetarKirov) [inline .travis.yml:2]: BTW, why is sudo required?

File: README.md [hunk 2]
```diff
@@ -16,15 +16,30 @@ to support online compilation of user code in a safe sandbox.
...
+    $ bstdin=$(printf 'Venus\nParis\nMontreal' | base64 -w0)
+    $ docker run --rm dlangtour/core-exec:dmd $bsource $bstdin
+    Venus
+    Paris
+    Montreal
+
+### Custom compiler arguments
+
+    $ bsource=$(echo 'void main() { import std.stdio; version(Foo) writeln("Hello World"); }' | base64 -w0)
+    $ DOCKER_FLAGS="-version=Foo" docker run -e DOCKER_FLAGS --rm dlangtour/core-exec:dmd $bsource
```
Comment (PetarKirov) [inline README.md:41]: Shouldn't that be: `... docker run -e $DOCKER_FLAGS ...` or am I missing something?

**Comment by wilzbach** [comment]
As I would like to pass `-color` to the docker image, I'm going to merge this now.

File: .travis.yml (hunk 1 above)
Comment (wilzbach) [reply .travis.yml:2]: It was there from the beginning: https://github.com/dlang-tour/core-rdmd/blob/master/.travis.yml

Comment (wilzbach) [reply .travis.yml:2]: But it's a good point & I tried to remove it.

File: README.md (hunk 2 above)
Comment (wilzbach) [reply README.md:41]: You need to explicitly tell docker which environment variables it should export to the container.

**Pull request merged**